pkg/internal/ebpf/capabilitytracer/capabilitytracer.go

```diff
@@ -0,0 +1,239 @@
+			err := p.bpfObjects.ValidPids.Put(uint32(i), uint64(segment))
+			if err != nil {
+				p.log.Error("Error setting up pid in BPF space, sizes of Go and BPF maps don't match", "error", err, "i", i)
+			}
+		}
+	}
+}
+
+func (p *Tracer) AllowPID(pid, ns uint32, svc *svc.Attrs) {
+	p.pidsFilter.AllowPID(pid, ns, svc, ebpfcommon.PIDTypeKProbes)
+	p.rebuildValidPids()
+	p.log.Info("Allowing PID", "pid", pid)
+}
+
+func (p *Tracer) BlockPID(pid, ns uint32) {
+	p.pidsFilter.BlockPID(pid, ns)
+	p.rebuildValidPids()
+	p.log.Info("Blocking PID", "pid", pid)
+}
+
+// AlreadyInstrumentedLib implements ebpf.Tracer.
+func (p *Tracer) AlreadyInstrumentedLib(uint64) bool {
+	return false
+}
+
+// Constants implements ebpf.Tracer.
+func (p *Tracer) Constants() map[string]any {
+	return nil
+}
+
+// GoProbes implements ebpf.Tracer.
+func (p *Tracer) GoProbes() map[string][]*ebpfcommon.ProbeDesc {
+	return nil
+}
+
+// ProcessBinary implements ebpf.Tracer.
+func (p *Tracer) ProcessBinary(*exec.FileInfo) {
+}
+
+// RecordInstrumentedLib implements ebpf.Tracer.
+func (p *Tracer) RecordInstrumentedLib(uint64, []io.Closer) {
+}
+
+// RegisterOffsets implements ebpf.Tracer.
+func (p *Tracer) RegisterOffsets(*exec.FileInfo, *goexec.Offsets) {
+}
+
+// SockMsgs implements ebpf.Tracer.
+func (p *Tracer) SockMsgs() []ebpfcommon.SockMsg {
+	return nil
+}
+
+// SockOps implements ebpf.Tracer.
+func (p *Tracer) SockOps() []ebpfcommon.SockOps {
+	return nil
+}
+
+// SocketFilters implements ebpf.Tracer.
+func (p *Tracer) SocketFilters() []*ebpf.Program {
+	return nil
+}
+
+// UProbes implements ebpf.Tracer.
+func (p *Tracer) UProbes() map[string]map[string][]*ebpfcommon.ProbeDesc {
+	return nil
+}
+
+// UnlinkInstrumentedLib implements ebpf.Tracer.
+func (p *Tracer) UnlinkInstrumentedLib(uint64) {
+}
+
+var _ beyla_ebpf.Tracer = (*Tracer)(nil)
+
+func New(cfg *beyla.Config, metrics imetrics.Reporter) *Tracer {
+	log := slog.With("component", "capabilitytracer.Tracer")
+	return &Tracer{
+		log:        log,
+		cfg:        cfg,
+		metrics:    metrics,
+		pidsFilter: ebpfcommon.CommonPIDsFilter(&cfg.Discovery),
+	}
+}
+
+func (p *Tracer) Load() (*ebpf.CollectionSpec, error) {
+	loader := loadBpf
+	if p.cfg.EBPF.BpfDebug {
+		loader = loadBpf_debug
+	}
+
+	return loader()
+}
+
+func (p *Tracer) BpfObjects() any {
+	return &p.bpfObjects
+}
+
+func (p *Tracer) AddCloser(c ...io.Closer) {
+	p.closers = append(p.closers, c...)
+}
+
+func (p *Tracer) KProbes() map[string]ebpfcommon.ProbeDesc {
+	kprobes := map[string]ebpfcommon.ProbeDesc{
+		"capable": {
+			Required: true,
+			Start:    p.bpfObjects.BeylaKprobeCapable,
+		},
+	}
+
+	return kprobes
+}
+
+func (p *Tracer) Tracepoints() map[string]ebpfcommon.ProbeDesc {
+	return nil
+}
+
+func (p *Tracer) SetupTailCalls() {}
+
+func (p *Tracer) Run(ctx context.Context, ch chan<- []request.Span) {
+	ebpfcommon.ForwardRingbuf(
+		&p.cfg.EBPF,
+		p.bpfObjects.CapabilityEvents,
+		&ebpfcommon.IdentityPidsFilter{},
+		// p.pidsFilter,
```

Comment:
If I use `p.pidsFilter` then the ring buffer filter rejects all the spans because the namespace in the PID struct is empty. What should this namespace be set to? I couldn't find an example in the code.